src/Apps/Sell/ArtworkFormContext.tsx

```diff
@@ -0,0 +1,109 @@
+import { UpdateSubmissionMutationInput } from "__generated__/UpdateConsignSubmissionMutation.graphql";
+import { createOrUpdateConsignSubmission } from "Apps/Consign/Routes/SubmissionFlow/Utils/createOrUpdateConsignSubmission";
+import { createContext, useContext, useEffect, useState } from "react";
+import { useRouter } from "System/Router/useRouter";
+import { useSystemContext } from "System/SystemContext";
+
+interface Actions {
+  goToPreviousStep: () => void
+  goToNextStep: () => void
+  updateSubmission: (values: UpdateSubmissionMutationInput) => void
+}
+
+interface State {
+  atFirstStep: boolean
+  atLastStep: boolean
+  currentStep: string
+  submissionID: string | undefined
+}
+interface ArtworkFormContextProps {
+  actions: Actions
+  state: State
+}
+
+export const ArtworkFormContext = createContext<ArtworkFormContextProps>({} as any)
+
+export const STEPS = [
+  "title",
+  "photos",
+  "details",
+  "purchase-history",
+  "dimensions"
+]
+
+interface ArtworkFormContextProviderProps {
+  children: React.ReactNode
+  submissionID?: string
+}
+
+export const ArtworkFormContextProvider: React.FC<ArtworkFormContextProviderProps> = ({ children, submissionID }) => {
+  const { match, router } = useRouter()
+  const { relayEnvironment } = useSystemContext()
+  const [currentStep, setCurrentStep] = useState<string>(match.location.pathname.split("/").pop() ?? "")
+  const [atFirstStep, setAtFirstStep] = useState<boolean>(false)
+  const [atLastStep, setAtLastStep] = useState<boolean>(false)
+
+  useEffect(() => {
+    const step = match.location.pathname.split("/").pop() ?? ""
+    setCurrentStep(step)
+    setAtFirstStep(step === STEPS[0])
+    setAtLastStep(step === STEPS[STEPS.length - 1])
+  }, [match.location])
```

Comment:
Is there something specific that we're looking for in `match.location`? That's a large object from found, so if it's a change in `pathname`, we should be specific about that.

Also, this would likely be fixed by @dzucconi's suggestion, but its a bit confusing that we're setting three separate steps like this, and on every change to match.location. What step are we at? Just worry about that. If something feels like it could be simpler, it likely can. Work from the ideal and go backwards 🙏

Reply:
Removed this hook as part of the state management update.